Mathlib/Data/Nat/Multiplicity.lean

```diff
@@ -272,22 +260,24 @@ theorem dvd_choose_pow (hp : Prime p) (hk : k ≠ 0) (hkp : k ≠ p ^ n) : p ∣
 
 theorem dvd_choose_pow_iff (hp : Prime p) : p ∣ (p ^ n).choose k ↔ k ≠ 0 ∧ k ≠ p ^ n := by
   refine' ⟨fun h => ⟨_, _⟩, fun h => dvd_choose_pow hp h.1 h.2⟩ <;> rintro rfl <;>
-    simpa [hp.ne_one] using h
+    simp [hp.ne_one] at h
 #align nat.prime.dvd_choose_pow_iff Nat.Prime.dvd_choose_pow_iff
 
 end Prime
 
-theorem multiplicity_two_factorial_lt : ∀ {n : ℕ} (h : n ≠ 0), multiplicity 2 n ! < n := by
+theorem multiplicity_two_factorial_lt : ∀ {n : ℕ} (_ : n ≠ 0), multiplicity 2 n ! < n := by
   have h2 := prime_two.prime
-  refine' binary_rec _ _
-  · contradiction
+  refine' binaryRec _ _
+  · exact fun h => False.elim <| h rfl
   · intro b n ih h
     by_cases hn : n = 0
     · subst hn
       simp at h
       simp [h, one_right h2.not_unit]
```

Comment:
This is a non-terminal simp, right? Please replace it by a `simp only`

Reply:
That must have been from mathlib3 but it is fixed now.